File: LeetCode/C++/1_Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> diffmap;
        vector<int> solution;
        for(int i=0;i<nums.size();i++) {
            diffmap[target-nums[i]] = i;
        }
        for(int j=0;j<nums.size();j++) {
            if(diffmap.find(nums[j])!=diffmap.end()) {
                 if(diffmap[nums[j]]!=j) {
                    solution.push_back(j);
                    solution.push_back(diffmap[nums[j]]);
                    return solution;
                 }
            }
        }
        return solution;
    }
};

File: LeetCode/Python/1_Two_Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        reference = {}
        answer = 0
        for idx, num in enumerate(nums):
            reference[num] = idx
        for idx, num in enumerate(nums):
            if target - num in reference and reference[target - num] != idx:
                return [idx, reference[target - num]]
        return -1